task8/src/main/java/PageServlet.java

```diff
@@ -0,0 +1,38 @@
+import adList.Ad;
+import adList.AdList;
+import adList.FilterConfig;
+import com.google.gson.Gson;
+
+import javax.servlet.*;
+import javax.servlet.http.*;
+import javax.servlet.annotation.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+@WebServlet(name = "PageServlet", value = "/ads/search")
+public class PageServlet extends HttpServlet {
+    AdList adList;
```

Comment:
The following code is exactly the same as in a previous servlet. Please always avoid copypasting. Create a common method, a factory class, or just make the two servlets both inherits the same superclass